packages/sources/coinpaprika-state/src/transport/coinpaprika-state.ts

```diff
@@ -0,0 +1,238 @@
+import { toNumber } from '../utils'
+import {
+  PairRequest,
+  SSEConnectionCallbacks,
+  SSEConnectionConfig,
+  SSEConnectionManager,
+} from './sse-connection-manager'
+
+const logger = makeLogger('CoinpaprikaStateTransport')
+
+const COINPAPRIKA_STATE_EVENT_TYPE = 't_s'
+
+type RequestParams = typeof inputParameters.validated
+
+// API returns numbers as strings, handle both types
+interface CoinpaprikaStreamData {
+  block_time: string | number
+  base_token_symbol: string
+  quote_symbol: string
+  volume_7d_usd: string | number
+  market_depth_plus_1_usd: string | number
+  market_depth_minus_1_usd: string | number
+  state_price: string | number
+  // Error fields from API
+  error?: string
+  details?: string
+}
+
+export type TransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: Array<{ base: string; quote: string }>
+    ResponseBody: CoinpaprikaStreamData
+  }
+}
+
+/**
+ * SSE transport that batches all pairs into one POST connection
+ */
+export class CoinpaprikaStateTransport extends StreamingTransport<TransportTypes> {
+  name!: string
+  responseCache!: ResponseCache<TransportTypes>
+  requester!: Requester
+
+  private connectionManager!: SSEConnectionManager
+  private activePairs: Map<string, RequestParams> = new Map()
+
+  async initialize(
+    dependencies: TransportDependencies<TransportTypes>,
+    adapterSettings: TransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this.requester = dependencies.requester
+    this.connectionManager = new SSEConnectionManager(this.requester, COINPAPRIKA_STATE_EVENT_TYPE)
+  }
+
+  async streamHandler(
+    context: EndpointContext<TransportTypes>,
+    subscriptions: SubscriptionDeltas<TypeFromDefinition<TransportTypes['Parameters']>>,
+  ): Promise<void> {
+    if (
+      subscriptions.new.length ||
+      subscriptions.stale.length ||
+      !this.connectionManager.connected
+    ) {
+      const currentPairs = new Map<string, RequestParams>()
+      for (const sub of subscriptions.desired) {
+        currentPairs.set(`${sub.base}/${sub.quote}`, sub)
+      }
+
+      logger.debug(`Updating stream: ${currentPairs.size} pairs`)
+      await this.updateConnection(context, currentPairs)
+    }
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  private async updateConnection(
+    context: EndpointContext<TransportTypes>,
+    pairs: Map<string, RequestParams>,
+  ): Promise<void> {
+    logger.debug(`updateConnection called with ${pairs.size} pairs`)
+
+    // update active pairs
+    this.activePairs = new Map(pairs)
+
+    if (pairs.size === 0) {
+      logger.debug('No pairs, disconnecting')
+      await this.connectionManager.disconnect()
+      return
+    }
+
+    const pairRequests: PairRequest[] = Array.from(pairs.values()).map((p) => ({
```

Comment:
This starts from the params list above, then builds a map, then converts back to a list in three different places